components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -111,6 +110,10 @@ void BraveWalletService::Bind(
 absl::optional<std::string> BraveWalletService::GetChecksumAddress(
     const std::string& contract_address,
     const std::string& chain_id) {
+  if (contract_address.empty()) {
```

Comment:
shouldn't we prevent assets other than eth from adding empty contract_address?

Reply:
```
diff --git a/components/brave_wallet/browser/brave_wallet_service.cc b/components/brave_wallet/browser/brave_wallet_service.cc
index 36adf975dd..6e2a64c641 100644
--- a/components/brave_wallet/browser/brave_wallet_service.cc
+++ b/components/brave_wallet/browser/brave_wallet_service.cc
@@ -214,6 +214,10 @@ void BraveWalletService::AddUserAsset(mojom::ERCTokenPtr token,
     std::move(callback).Run(false);
     return;
   }
+  if (optional_checksum_address->empty() && token->symbol != "ETH") {
+    std::move(callback).Run(false);
+    return;
+  }
   const std::string checksum_address = optional_checksum_address.value();

   const std::string network_id = GetNetworkId(prefs_, chain_id);

```

Reply:
Fixed in https://github.com/brave/brave-core/pull/10333/commits/970f83b682abfe55a9ff473a9d5af166c8ec5130